core/src/main/resources/error/error-classes.json

```diff
@@ -0,0 +1,22 @@
+{
+  "DIVIDE_BY_ZERO_ERROR" : {
+    "sqlState" : "22012",
+    "messageFormatLines" : [ "divide by zero" ]
+  },
+  "DUPLICATE_KEY_ERROR" : {
+    "sqlState" : "23000",
+    "messageFormatLines" : [ "Found duplicate keys '%s'" ]
+  },
+  "MISSING_COLUMN_ERROR" : {
+    "sqlState" : "42000",
+    "messageFormatLines" : [ "cannot resolve '%s' given input columns: [%s]" ]
+  },
+  "SECOND_FUNCTION_ARGUMENT_NOT_INTEGER_ERROR" : {
+    "sqlState" : "22023",
+    "messageFormatLines" : [ "The second argument of '%s' function needs to be an integer." ]
+  },
+  "WRITING_JOB_ABORTED_ERROR" : {
+    "sqlState" : "40000",
+    "messageFormatLines" : [ "Writing job aborted" ]
+  }
+}
```

Comment:
Do you mean `sqlState` always unique?

Reply:
Not necessarily; `sqlState` can be re-used, especially for common error classes without known subclasses, like `42000` (syntax or semantic error). In those cases, the error class will be the disambiguator.